Also, the OSX build is failing - https://travis-ci.org/github/microsoft/AirSim/jobs/670441638#L6866
```
ld: library not found for -lc++fs
clang-8: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [output/bin/MavLinkTest] Error 1
```

As to why the build isn't showing up as failed, my thinking is the root `build.sh` exits on failure (`set -e`), but Travis only checks the return code of the last command being run, which would be the Unity build.
So the `build.sh` exits with code 1, Travis continues with the `echo` and Unity build, which is where things get confusing, there's no output after that, which might indicate some problem in the beginning of the script itself, since everything goes into the `stderr` rather than `stdout` and hence is captured in the file, but the word `error:` was not found, and so no output. But then why isn't `exit 1` giving a return code of 1??
In the Unity build command, `||` is used to check the exit code of the command on the left side and running the right only if the left one exits with a non-zero code. I've tested this on my local system with some simple bash scripts, so am not sure why it's behaving in this way

One addition to make Travis exit if the root `build.sh` exits with failure is to add `|| travis_terminate 1` to it, so that it doesn't continue to the Unity build

--- a/.travis.yml
+++ b/.travis.yml
@@ -39,7 +39,7 @@ before_install:
 script:
   - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
       ./setup.sh;
-      ./build.sh;
+      ./build.sh || travis_terminate 1;
       echo "Starting Unity Build!";
       cd Unity && ./build.sh 2> Unity_error.txt || ((cat Unity_error.txt | grep -i "error:" -B 10 -A 10) && exit 1);
     elif [[ "$TRAVIS_OS_NAME" == "windows" ]]; then
@@ -49,7 +49,7 @@ script:
       cd Unity '&&' build.cmd;
     elif  [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
       ./setup.sh;
-      ./build.sh;
+      ./build.sh || travis_terminate 1;
       echo "Starting Unity Build!";
       cd Unity && ./build.sh 2> Unity_error.txt || ((cat Unity_error.txt | grep -i "error:" -B 10 -A 10) && exit 1);
     fi

--- a/.travis.yml
+++ b/.travis.yml
@@ -21,6 +21,7 @@ jobs:
 
     - name: MacOS
       os: osx
+      osx_image: xcode11.3
 
 
 before_install:

--- a/cmake/cmake-modules/CommonSetup.cmake
+++ b/cmake/cmake-modules/CommonSetup.cmake
@@ -48,10 +48,9 @@ macro(CommonSetup)
     IF(UNIX)
         set(RPC_LIB_DEFINES "-D MSGPACK_PP_VARIADICS_MSVC=0")
         set(BUILD_TYPE "linux")
-
-        if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang")
+        if (APPLE)
             set(CMAKE_CXX_STANDARD 17)
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D__CLANG__")
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wstrict-aliasing -D__CLANG__")
         else ()
             set(CMAKE_CXX_FLAGS "\
                 -std=c++17 -stdlib=libc++ -ggdb -Wall -Wextra -Wstrict-aliasing -Wunreachable-code -Wcast-qual -Wctor-dtor-privacy \

